### PR TITLE
Locale negotiation

### LINKED ISSUE #1958: Locale negotiation
Add locale to the URL based on the browser's language preferences.

### PR DESCRIPTION
**Pull request by alanmoo**
Fixes #1958

**Changes proposed in this pull request:**
- Automatically redirects users to appropriate locale (via mofo-localize module)
- Update routes to support locales
- Update links to point to current locale url

I believe I've gotten everything, but it's hard to test every link on the site

### CODE REVIEW

**Comment by alanmoo** [comment]
This is still failing and I don't know why. @pomax or @alicoding can you help?

**Comment by alicoding** [comment]
If you have any ES6 make it ES5 since it's not enabled here...

**Comment by alanmoo** [comment]
Just when I finally got used to typing ` for strings...

**Comment by alicoding** [comment]
Just assigning @Pomax to help review this one as well if he got time.

File: app.js [hunk 1]
```diff
@@ -25,6 +26,8 @@ var indexStatic;
 var router;
 var matcher;
 var app = express();
+var locale = "";
```
Comment (alicoding) [inline app.js]: Let's default the locale to `en-US`

File: app.js [hunk 2]
```diff
@@ -209,6 +212,13 @@ app.use(function(req, res, next) {
 
     // if this belongs to one of the predefined urls, let's generate its associated page
     if ( matcher.match(location) ) {
+      var search = url.parse(req.url).search || "";
+
+      locale = localize.parseLocale(req.headers["accept-language"], location, locales).locale;
```
Comment (alicoding) [inline app.js]: what does `parseLocale` return here? Just curious

**Comment by alicoding** [comment]
This looks good to me, but can you add test for query preserved here? I know it's working, but just want to prevent it from breaking in the future.

**Comment by alanmoo** [comment]
@alicoding Added it, lmk if that's what you were looking for.

**Comment by alicoding** [comment]
R+ from me

**Comment by alanmoo** [comment]
Gonna merge this in now because it's blocking @alicoding but @pomax maybe you can give it a once over tomorrow and I submit a follow up.

**Pull request merged**